launch/launcher_test.go

```diff
@@ -233,15 +260,197 @@ func testLauncher(t *testing.T, when spec.G, it spec.S) {
-					h.AssertEq(t, syscallExecArgsColl[0].argv, []string{"notepad", "arg1", "arg2"})
+					h.AssertEq(t, syscallExecArgsColl[0].argv, []string{"notepad", "arg1", "arg with space"})
 				} else {
 					h.AssertEq(t, syscallExecArgsColl[0].argv0, "/bin/sh")
-					h.AssertEq(t, syscallExecArgsColl[0].argv, []string{"sh", "arg1", "arg2"})
+					h.AssertEq(t, syscallExecArgsColl[0].argv, []string{"sh", "arg1", "arg with space"})
 				}
 				h.AssertEq(t, syscallExecArgsColl[0].envv, envList)
 			})
 		})
 
+		when("a start command is NOT marked as direct", func() {
+			when("the providing buildpack has api >= 0.4", func() {
+				it.Before(func() {
+					launcher.Buildpacks = []launch.Buildpack{
+						{API: "0.4", ID: "bp.1"},
+					}
+					launcher.Processes = []launch.Process{{
+						Type:        "some-type",
+						Command:     "some-command",
+						BuildpackID: "bp.1",
+					}}
+					launcher.DefaultProcessType = "some-type"
+				})
+
+				when("there are one or more args", func() {
+					it.Before(func() {
+						launcher.Processes[0].Args = []string{"arg1", "arg with space"}
+					})
+
+					it("treats command and args as bash command tokens", func() {
+						if err := launcher.Launch("/path/to/launcher", nil); err != nil {
+							t.Fatal(err)
+						}
+
+						if len(syscallExecArgsColl) != 1 {
+							t.Fatalf("expected syscall.Exec to be called once: actual %v\n", syscallExecArgsColl)
+						}
+
+						if runtime.GOOS == "windows" {
+							h.AssertEq(t, syscallExecArgsColl[0].argv0, "cmd")
+							h.AssertEq(t, syscallExecArgsColl[0].argv, []string{
+								"cmd", "/q", "/c", `some-command`, `arg1`, "arg with space",
+							})
+						} else {
+							h.AssertEq(t, syscallExecArgsColl[0].argv0, "/bin/bash")
+							h.AssertEq(t, syscallExecArgsColl[0].argv, []string{
+								"bash", "-c",
+								`exec bash -c '"$(eval echo \"$0\")" "$(eval echo \"$1\")" "$(eval echo \"$2\")"' "${@:1}"`,
+								"/path/to/launcher", "some-command", "arg1", "arg with space",
+							})
+						}
+					})
+				})
+
+				when("there are zero args", func() {
+					it("treats the command as a bash script", func() {
+						if err := launcher.Launch("/path/to/launcher", nil); err != nil {
+							t.Fatal(err)
+						}
+
+						if len(syscallExecArgsColl) != 1 {
+							t.Fatalf("expected syscall.Exec to be called once: actual %v\n", syscallExecArgsColl)
+						}
+
+						if runtime.GOOS == "windows" {
+							h.AssertEq(t, syscallExecArgsColl[0].argv0, "cmd")
+							h.AssertEq(t, syscallExecArgsColl[0].argv, []string{
+								"cmd", "/q", "/c", "some-command",
+							})
+						} else {
+							h.AssertEq(t, syscallExecArgsColl[0].argv0, "/bin/bash")
+							h.AssertEq(t, syscallExecArgsColl[0].argv, []string{
+								"bash", "-c",
+								`exec bash -c "$@"`,
+								"/path/to/launcher", "some-command",
+							})
+						}
+					})
+				})
+			})
+
+			when("buildpacks have provided profile.d scripts", func() {
```

Comment:
Is this just a nesting change?

Reply:
Yes, I moved it under `when("a start command is NOT marked as direct", func() {}` to make it clear that profile scripts are only executed for `direct=false` processes.